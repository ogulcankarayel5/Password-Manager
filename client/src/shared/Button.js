import {Link} from 'react-router-dom'
import styled from "styled-components";

export const Button = styled(Link)`
  font-size:${props=>props.fontSize ?? '2rem'};
  color:${props=>props.theme.colors.background};
  background-color:${props=>props.color? props.color : props.theme.colors.primary};
  padding:${props=>props.padding ?? '2rem'};
  border-radius: 25%/50%;
  font-weight:${props=>props.fontWeight ?? 500};
  cursor:pointer;
  

`;

Button.displayName='Button'