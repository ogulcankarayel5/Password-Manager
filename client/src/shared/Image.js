import styled from "styled-components";

export const ResponsiveImage = styled.img.attrs({
  
    alt:"img"
  })
 `
   height:auto;
   max-width:100%;
 `
 
 