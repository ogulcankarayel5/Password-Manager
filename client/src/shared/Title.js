import styled from "styled-components";


export const Title = styled.h1`
font-size: 6rem;
color: ${(props) => props.theme.colors.mainTitle};
`;
