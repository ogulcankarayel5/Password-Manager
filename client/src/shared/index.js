export * from './Button'
export * from './Container'
export * from './Image'
export * from './Text'
export * from './Theme'
export * from './Title'