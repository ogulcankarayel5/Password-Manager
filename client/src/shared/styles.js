import styled,{css} from "styled-components";



const defaultFlex = css`
  display: flex;
  align-items: center;
  justify-content: center;
`;


export {defaultFlex};

