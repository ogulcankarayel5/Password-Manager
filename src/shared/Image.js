import styled from "styled-components";
import { LazyLoadImage } from 'react-lazy-load-image-component';
export const ResponsiveImage = styled(LazyLoadImage)`
 height:auto;
   max-width:100%;

`;
 
 