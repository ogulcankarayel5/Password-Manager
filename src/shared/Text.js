import styled from "styled-components";


export const Text = styled.p`
  font-size: 2rem;
  color: ${(props) => props.theme.colors.text};
  font-weight: 600;
  line-height:1.5;
  margin-top:${props=>props.marginTop ? props.marginTop : 0};
  text-align:${props=>props.textStart ? "start" : "center"};
`;
