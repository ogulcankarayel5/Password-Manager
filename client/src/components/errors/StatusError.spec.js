import React from "react";
import { shallow} from "enzyme";
import { StatusError, StatusErrorDiv } from "./";
import { Button } from "../../shared";

import { theme } from "../../shared";
import renderer from "react-test-renderer";
import "jest-styled-components";

const setUp = (
  props = { title: 404, description: "Not found", color: theme.colors.statusErrorBackground }
) => {
  const component = shallow(<StatusError {...props} />);
  return component;
};
// describe("Status error component",()=>{
//     it('matches the snapshot',()=>{
//         const tree=renderer.create(

//                 <StatusError/>

//         ).toJSON()

//         expect(tree).toMatchSnapshot()
//     })
// })

describe("StatusError component", () => {
  let component;

  beforeEach(() => {
    component=setUp();
  });

  afterEach(() => {
    component.unmount();
  });

  it("it should render one button", () => {
    console.log(component.debug());
    const wrapper = component.find(Button);
    console.log(wrapper);
    expect(wrapper.length).toBe(1);
  });
});




describe("StatusErrorDiv styled component snapshot testing", () => {
  it("Snapshot for StatusErrorDiv", () => {
    const tree = renderer.create(<StatusErrorDiv theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
