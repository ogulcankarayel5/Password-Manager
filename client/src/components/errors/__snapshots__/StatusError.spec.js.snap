// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusErrorDiv styled component snapshot testing Snapshot for StatusErrorDiv 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: url(p404.png), #151729;
  box-shadow: 0 15px 30px rgba(0,0,0,0.5);
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 20rem;
  max-width: 600px;
  text-align: center;
}

.c0 h2 {
  margin-bottom: 3rem;
  font-size: 5rem;
  color: #FFFFFF;
  line-height: 1rem;
}

.c0 p {
  color: #FFFFFF;
  font-size: 2rem;
}

.c0 a {
  margin-top: 2.5rem;
  color: #FFFFFF;
  border-radius: 25px;
}

<div
  className="c0"
/>
`;
